lightning/src/ln/channelmanager.rs

```diff
@@ -6097,6 +6347,27 @@ impl<'a, Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 	}
 }
 
+fn hkdf_extract_expand(salt: &[u8], ikm: &KeyMaterial) -> ([u8; 32], [u8; 32], [u8; 32]) {
```

Comment:
Could this be a constructor for `ExpandedPmtKey` instead?

Reply:
My thinking is that as is, it's a generic crypto util, whereas `ExpandedPmtKey` is specific to inbound payments. Fyi it's basically duplicated in `peer_channel_encryptor.rs` -- as such, seems it'd ideally belong in some generic `crypto_utils` module. Not sure if worth doing now though :thinking: